sshd-core/src/main/java/org/apache/sshd/server/channel/ChannelSession.java

```diff
@@ -908,7 +908,7 @@ protected void closeShell(int exitValue, boolean closeImmediately) throws IOExce
 
         if (!isClosing()) {
             if (out != null) {
-                out.flush();
```

Comment:
Not sure about this... "smells fishy" as well , though I cannot put my fiinger on it - will have to think about it.

Reply:
`out.close()` works because this is indeed the last use of this stream. Looks fine to me...

Unrelated to this change:

But the code is a bit strange anyway. The intent is obviously to get all data from the command sent before the EOF is sent. But why not also flush or close `err`? What happens if the command writes something to the error stream? `err` gets closed only in the `close()` call in line 916, so that would flush it after the EOF (and the exit status) was sent. That's probably not OK.

And anyway: stderr is traditionally "unbuffered", i.e., flushed on newlines. I don't see anything in ChannelOutputStream that would implement that for SSH_MSG_CHANNEL_EXTENDED_DATA. Maybe that's OK; the command would be responsible for flushing it frequently then if desired.

Reply:
> out.close() works because this is indeed the last use of this stream. Looks fine to me...

I trust your judgment @tomaswolf 

>  The intent is obviously to get all data from the command sent before the EOF is sent. But why not also flush or close err? What happens if the command writes something to the error stream? err gets closed only in the close() call in line 916, so that would flush it after the EOF (and the exit status) was sent. That's probably not OK.

I tend to agree with you...